src/ess/amor/conversions.py

```diff
@@ -60,10 +61,10 @@ def theta(wavelength, divergence_angle, L2, sample_rotation, detector_rotation):
     '''
     c = sc.constants.g * sc.constants.m_n**2 / sc.constants.h**2
     out = (c * L2 * wavelength**2).to(unit='dimensionless') + sc.sin(
-        divergence_angle + detector_rotation
+        divergence_angle.to(unit='rad') + detector_rotation.to(unit='rad')
     )
     out = sc.asin(out, out=out)
-    out -= sample_rotation
+    out -= sample_rotation.to(unit='rad')
```

Comment:
Are the variables you are converting all scalar? For arrays, I would use `copy=False`.